client/transaction-pool/src/lib.rs

```diff
@@ -198,6 +217,10 @@ impl<PoolApi, Block> TransactionPool for BasicPool<PoolApi, Block>
 	fn ready_transaction(&self, hash: &TxHash<Self>) -> Option<Arc<Self::InPoolTransaction>> {
 		self.pool.ready_transaction(hash)
 	}
+
+	fn background_task(&self) -> Option<Pin<Box<dyn Future<Output = ()> + Send>>> {
```

Comment:
You allowed to call it any number of times, it will return the actual task only once. I don't see other way to do it within the current `ServiceBuilder`